src/TropicalGeometry/variety_prime.jl

```diff
@@ -19,16 +19,19 @@ function tropical_variety_prime(I::MPolyIdeal, nu::TropicalSemiringMap; weighted
     return dehomogenize_post_tropicalization(TropIh)
 end
 
+
 # trivial valuation
 function tropical_variety_prime_singular(I::MPolyIdeal, nu::TropicalSemiringMap{QQField,Nothing,<:Union{typeof(min),typeof(max)}}; weighted_polyhedral_complex_only::Bool=false)
-    R = base_ring(I)
-    singularCommand = join(["ring r=0,("*join(string.(symbols(R)),",")*"),dp;",
-                            "ideal I = "*join(string.(gens(I)), ",")*";",
-                            "if (!defined(tropicalVariety)) { LIB \"tropical.lib\"; };",
-                            "fan TropI = tropicalVariety(I);",
-                            "string TropIString = string(TropI);"])
-    Singular.call_interpreter(singularCommand)
-    TropIString = Singular.lookup_library_symbol("Top", "TropIString")
+
+    name = "tropicalVariety_as_string"
+    sI = Oscar.singular_generators(I)
+    arguments = Any[ Singular.prepare_argument(sI)[1] ]
+    return_value = Singular.libSingular.call_singular_library_procedure(name, base_ring(sI).ptr, arguments);
+    if Singular.libSingular.have_error()
+        error(Singular.libSingular.get_and_clear_error())
+    end
+    TropIString = Singular.convert_return(return_value, ring)
```

Comment:
I think this should also work, as the first "libname" argument to `Singular.low_level_caller` is only used to ensure that library is loaded (which is the case here anyway, so it basically does nothing), and then the lookup of the second argument (funcname) ignores the library name.
```suggestion
    sI = Oscar.singular_generators(I)
    TropIString = Singular.low_level_caller("tropical", "tropicalVariety_as_string", sI)
```

However, I did not test it. But if it works, it is clearly preferable :-)

Reply:
Seems to be working.

Reply:
@fingolfin do you happen to know how to call the following Singular function using `Singular.low_level_caller` for two inputs, the ideal `sI` and an integer `2`:

```
Singular.call_interpreter("proc tropicalVariety_as_string(ideal I, list #) { if(size(#)==0) { return(string(tropicalVariety(I))); }; return(string(tropicalVariety(I,#[1]))); }")
```

For the old version using `Singular.libSingular.call_singular_library_procedure` the following arguments worked:
```
arguments = Any[ Singular.prepare_argument(sI)[1], Singular.prepare_argument(Int(p))[1] ]
```

Reply:
Nvm, I take back my comment that `arguments` above worked.  It works for a dummy function:

```
julia> Singular.call_interpreter("proc foo2(ideal I, list #) { print(#[1]); }") # function to call
4-element Vector{Any}:
 false
      ""
      ""
      ""

julia> I = grassmann_pluecker_ideal(2, 5); # argument 1

julia> p = ZZ(3); # argument 2

julia> sI = Oscar.singular_generators(I);

julia> sR = base_ring(sI)
Singular polynomial ring (QQ),(x_1,x_2,x_3,x_4,x_5,x_6,x_7,x_8,x_9,x_10),(dp(10),C)

julia> sp = Int(p)
3

julia> arguments = Any[ Singular.prepare_argument(sI)[1], Singular.prepare_argument(sp)[1] ]
2-element Vector{Any}:
 Any[276, Ptr{Nothing}(0x000000001176b2a0)]
 Any[420, Ptr{Nothing}(0x0000000000000003)]

julia> return_value = Singular.libSingular.call_singular_library_procedure("foo2", sR.ptr, arguments);
3

```

But somehow not for `tropicalVariety_as_string` which returns an empty string:

```
julia> Singular.libSingular.load_library("tropical.lib")
true

julia> Singular.call_interpreter("proc tropicalVariety_as_string(ideal I, list #) { if(size(#)==0) { return(string(tropicalVariety(I))); }; return(string(tropicalVariety(I,#[1]))); }") # function to call
4-element Vector{Any}:
 false
      ""
      ""
      ""

julia> I = grassmann_pluecker_ideal(2, 5); # argument 1

julia> nu = tropical_semiring_map(QQ, 3)
Map into Min tropical semiring encoding the 3-adic valuation on Rational field

julia> p = Oscar.uniformizer_ring(nu) # argument 2
3

julia> sI = Oscar.singular_generators(I);

julia> sR = base_ring(sI)
Singular polynomial ring (QQ),(x_1,x_2,x_3,x_4,x_5,x_6,x_7,x_8,x_9,x_10),(dp(10),C)

julia> sp = Int(p)
3

julia> arguments = Any[ Singular.prepare_argument(sI)[1], Singular.prepare_argument(sp)[1] ]
2-element Vector{Any}:
 Any[276, Ptr{Nothing}(0x0000000011753af8)]
 Any[420, Ptr{Nothing}(0x0000000000000003)]

julia> return_value = Singular.libSingular.call_singular_library_procedure("tropicalVariety_as_string", sR.ptr, arguments);

julia> return_value
3-element Vector{Any}:
 false
      Ptr{Nothing}(0x0000000011741750)
   511

julia> TropIString = Singular.convert_return(return_value, ring)
""

```

I will reinstate the old version for the second function, since at least it did work.